src/screens/DiscoveryScreenV2/fragment/DomainFragment.js

```diff
@@ -1,28 +1,74 @@
+import {useNavigation} from '@react-navigation/native';
 import PropTypes from 'prop-types';
+import * as React from 'react';
 import {FlatList, Keyboard, StyleSheet, Text, View} from 'react-native';
-import {useNavigation, useRoute} from '@react-navigation/native';
 
+import Accordion from 'react-native-collapsible/Accordion';
+import MemoDomainProfilePictureEmptyState from '../../../assets/icon/DomainProfilePictureEmptyState';
+import LoadingWithoutModal from '../../../components/LoadingWithoutModal';
+import {Context} from '../../../context/Store';
 import DiscoveryAction from '../../../context/actions/discoveryAction';
-import DiscoveryTitleSeparator from '../elements/DiscoveryTitleSeparator';
-import DomainList from '../elements/DiscoveryItemList';
 import FollowingAction from '../../../context/actions/following';
-import LoadingWithoutModal from '../../../components/LoadingWithoutModal';
-import MemoDomainProfilePictureEmptyState from '../../../assets/icon/DomainProfilePictureEmptyState';
-import RecentSearch from '../elements/RecentSearch';
-import dimen from '../../../utils/dimen';
 import useIsReady from '../../../hooks/useIsReady';
-import {COLORS} from '../../../utils/theme';
-import {Context} from '../../../context/Store';
 import {followDomain, unfollowDomain} from '../../../service/domain';
+import dimen from '../../../utils/dimen';
 import {fonts} from '../../../utils/fonts';
-import {getUserId} from '../../../utils/users';
+import {COLORS} from '../../../utils/theme';
+import DomainList from '../elements/DiscoveryItemList';
+import DiscoveryTitleSeparator from '../elements/DiscoveryTitleSeparator';
+import RecentSearch from '../elements/RecentSearch';
 
 const FROM_FOLLOWED_DOMAIN = 'fromfolloweddomains';
 const FROM_FOLLOWED_DOMAIN_INITIAL = 'fromfolloweddomainsinitial';
 const FROM_UNFOLLOWED_DOMAIN = 'fromunfolloweddomains';
 
+const SECTIONS = [
+  {
+    title: 'First',
+    content: 'Lorem ipsum...'
+  }
+];
+
+const AccordionView = ({data, renderItem, setActiveSections, activeSections}) => {
+  const renderSectionTitle = () => {
+    return <View style={styles.content}></View>;
+  };
+
+  const renderHeader = (data, index) => {
+    return (
+      <DiscoveryTitleSeparator
+        withBorderBottom={true}
+        key="user-title-separator"
+        text="Domains you follow"
+        showArrow
+        rotateArrow={activeSections?.some((actived) => actived === index)}
+      />
+    );
+  };
+
+  const renderContent = () => {
+    return (
+      <View style={styles.content}>{data?.map((item, index) => renderItem({index, item}))}</View>
+    );
+  };
+
+  const updateSections = (activeSectionsParams) => {
+    setActiveSections(activeSectionsParams);
+  };
+
+  return (
+    <Accordion
+      sections={SECTIONS}
+      activeSections={activeSections}
+      renderSectionTitle={renderSectionTitle}
+      renderHeader={renderHeader}
+      renderContent={renderContent}
+      onChange={updateSections}
+    />
+  );
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The AccordionView component is introduced for collapsible sections. It's a good practice to break down complex components into smaller, more manageable ones. However, ensure that the `renderItem` function passed as a prop to this component handles null or undefined data gracefully.
```javascript
const AccordionView = ({data, renderItem, setActiveSections, activeSections}) => {
  // ...
};
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->